backend/validators/views.py

```
@@ -614,12 +616,270 @@ def _invalidate_wall_of_shame_cache(cls):
+            context['grace_days_remaining'] = max(
+                0,
+                cls.VERSION_SHAME_GRACE_DAYS - context['target_elapsed_days'],
+            )
+            return context
+
+        context['status'] = 'shame'
+        context['shame_started_at'] = target_date + timedelta(days=cls.VERSION_SHAME_GRACE_DAYS)
+        return context
+
+    @classmethod
+    def _sync_shame_started_at(cls, wallets, targets, now):
```

Ruff (0.15.15)

[warning] Missing return type annotation for classmethod `_sync_shame_started_at`

Add return type annotation: `None`

(ANN206)